### PR TITLE
TSK-1841: refactor GetClassificationAccTest to use test-api

### PR DESCRIPTION
**Pull request by ryzheboka**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    assertThat(classification)
+        .hasNoNullFieldsOrProperties()
+        .isEqualTo(classificationWithAllProperties);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ThrowException_When_ClassificationIdIsNull() {
+    assertThatThrownBy(() -> classificationService.getClassification(null))
+        .isInstanceOf(ClassificationNotFoundException.class);
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:81]:
```suggestion
    ThrowingCallable test = () -> classificationService.getClassification(null);
    
    assertThatThrownBy(test).isInstanceOf(ClassificationNotFoundException.class);
```

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ThrowException_When_ClassificationIdDoesNotExist() {
+    assertThatThrownBy(() -> classificationService.getClassification("CLI:NonExistingId"))
+        .isInstanceOf(ClassificationNotFoundException.class);
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:88]: split

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_getClassificationByKeyAndDomain_When_AllPropertiesSet() throws Exception {
+    Classification classification =
+        classificationService.getClassification(
+            classificationWithAllProperties.getKey(), classificationWithAllProperties.getDomain());
+
+    assertThat(classification).isEqualTo(classificationWithAllProperties);
+  }
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:99]: move next to the test which gets the classification by id.
Question: Why don't you verify here that all properties are set?

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ThrowException_When_ClassificationKeyIsNull() {
+    assertThatThrownBy(
+            () ->
+                classificationService.getClassification(
+                    null, classificationWithAllProperties.getDomain()))
+        .isInstanceOf(ClassificationNotFoundException.class);
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:108]: split

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ThrowException_When_ClassificationKeyDoesNotExist() {
+    assertThatThrownBy(
+            () ->
+                classificationService.getClassification(
+                    "WrongKey", classificationWithAllProperties.getDomain()))
+        .isInstanceOf(ClassificationNotFoundException.class);
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:118]: split

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ReturnFromMasterDomain_When_ClassificationDomainDoesNotExist() throws Exception {
+    Classification classification =
+        classificationService.getClassification(
+            classificationWithAllProperties.getKey(), "WrongDomain");
+
+    assertThat(classification).isNotNull().hasNoNullFieldsOrProperties();
+    assertThat(classification.getDomain()).isEmpty();
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:129]:
```suggestion
    assertThat(classification.getDomain()).isEqualTo(MASTER_DOMAIN);
```

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    assertThatThrownBy(
+            () ->
+                classificationService.getClassification(
+                    "WrongKey", classificationWithAllProperties.getDomain()))
+        .isInstanceOf(ClassificationNotFoundException.class);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ReturnFromMasterDomain_When_ClassificationDomainDoesNotExist() throws Exception {
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:123]:
```suggestion
  void should_ReturnMasterClassification_When_ClassificationDomainDoesNotExist() throws Exception {
```

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  void should_GetClassification_When_NameContainsSpecialCharacter() throws Exception {
+    Classification classification =
+        DefaultTestEntities.defaultTestClassification()
+            .name("Zustimmungserklärung")
+            .buildAndStore(classificationService, "businessadmin");
+
+    Classification result = classificationService.getClassification(classification.getId());
+
+    assertThat(result.getName()).isEqualTo("Zustimmungserklärung");
+  }
```
Comment (mustaphazorgati) [inline lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:144]: what are we exactly testing here? That the unicode character gets set properly?
if so make that clear by its method name?
exaple:
should_GetSpecialCharacterCorretly_When_AFieldContainsASpecialCharacter

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java (hunk 2 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:88]: Ok :)

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java (hunk 5 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:118]: Ok :)

File: lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java (hunk 8 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/classification/get/GetClassificationAccTest.java:144]: I will rename the method

**Review by mustaphazorgati — DISMISSED** [review]

**Review by mustaphazorgati — APPROVED** [review]
